### ISSUE TITLE
ty exits with code 0 even if issues are detected

### ISSUE THREAD

**Issue by pawamoy** [astral-sh/ty#2680]
### Summary

I was expecting ty to exit with an error code greather than 0 when issues are detected within the code. But currently it looks like it always exits with 0? 🤔 

```console
% . .venvs/3.10/bin/activate
% type ty
ty is /run/media/pawamoy/Data/dev/copier-uv/tests/tmp/.venvs/3.10/bin/ty
% ty check .
warning[unused-ignore-comment]: Unused `ty: ignore` directive
  --> scripts/gen_credits.py:23:30
   |
21 |     import tomllib
22 | else:
23 |     import tomli as tomllib  # ty: ignore[unresolved-import]
   |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
24 |
25 | project_dir = Path(os.getenv("MKDOCS_CONFIG_DIR", "."))
   |
help: Remove the unused suppression comment
20 | if sys.version_info >= (3, 11):
21 |     import tomllib
22 | else:
   -     import tomli as tomllib  # ty: ignore[unresolved-import]
23 +     import tomli as tomllib
24 | 
25 | project_dir = Path(os.getenv("MKDOCS_CONFIG_DIR", "."))
26 | with project_dir.joinpath("pyproject.toml").open("rb") as pyproject_file:

Found 1 diagnostic
% echo $?
0
```

Shouldn't it exit with an error code of 1 in that case?

### Version

0.0.14

**Comment by pawamoy** [comment]
Ah my bad, found the relevant docs: https://docs.astral.sh/ty/reference/exit-codes/

> --error-on-warning: ty will exit with 1 if it finds any violations with severity warning or higher.
